### ISSUE TITLE
[Bug] Crash on Samsung Devices

### ISSUE THREAD

**Issue by hyplant** [BinTianqi/OwnDroid#85]
If Dhizuku is enabled, the app crashes while opening. Clear the app data, reopen it, and then activate Dhizuku, it will work normally.
If you close Dhizuku before exiting the app, you can still open and use it next time. If it is not closed, it will crash the next time it is opened.
  
logs:
```Text
时间：2024-11-24 15:12:33
类名：com.junge.algorithmAidePro.hook.MyUncaughtExceptionHandler
日志名称：捕获到异常(已拦截闪退)

返回结果类型：void
返回结果值：void

调用堆栈：
    at com.rosan.dhizuku.api.Dhizuku.getOwnerComponent(Dhizuku.java:148)
    at com.rosan.dhizuku.api.Dhizuku.getOwnerPackageName(Dhizuku.java:144)
    at com.bintianqi.owndroid.dpm.DPMKt.isDeviceOwner(DPM.kt:55)
    at com.bintianqi.owndroid.dpm.DPMKt.setDefaultAffiliationID(DPM.kt:412)
    at com.bintianqi.owndroid.MainActivity$onCreate$1.invokeSuspend(MainActivity.kt:102)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:367)
    at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
    at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:25)
    at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:110)
    at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
    at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
    at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
    at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
    at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
    at com.bintianqi.owndroid.MainActivity.onCreate(MainActivity.kt:102)
    at android.app.Activity.performCreate(Activity.java:8290)
    at android.app.Activity.performCreate(Activity.java:8270)
    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8663)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)

```

**Comment by Minoricew** [comment]
> If Dhizuku is enabled, the app crashes while opening. Clear the app data, reopen it, and then activate Dhizuku, it will work normally. If you close Dhizuku before exiting the app, you can still open and use it next time. If it is not closed, it will crash the next time it is opened.
> 
> logs:

Some necessary information is not provided in your error logs, which can make debugging more difficult:

- `ErrorMessage` (e.g. `Didn't find class "com.android.rickroll.MyClass" on path: DexPathList[...]`)
- `ErrorType` (e.g. `java.lang.ClassNotFoundException`) 

To get complete logs when errors occur, please DO NOT use third-party log viewing tools. Instead, use [LogFox](https://github.com/F0x1d/LogFox) or [AppErrorsTracking](https://github.com/KitsunePie/AppErrorsTracking).

**Comment by BinTianqi** [comment]
I fix it in eadb2e9, but my project couldn't be compiled in GitHub Action. I don't know why. I can compile this project on my computer.
